pkg/controller/clustercontroller/backups.go

```diff
@@ -37,6 +37,10 @@ var (
 	backupPollingTime = time.Second
 	// time to wait for a backup to be completed
 	backupWatchTimeout = time.Hour
+	// slice that contains backups
+	listBackups []*api.MysqlBackup
```

Comment:
Do not keep a global list of backup. You need to cleanup backups per cluster. That means that each `Run()`, after a backup is created, the old ones should get deleted.